skrf/multiNoisyNetworkSystem.py

```diff
@@ -0,0 +1,421 @@
+# -*- coding: utf-8 -*-
+"""
+.. module:: skrf.multiNoisyNetworkSystem
+=============================================================
+multiNoisyNetworkSystem (:mod:`skrf.multiNoisyNetworkSystem`)
+=============================================================
+
+Tools for combining multiple :class:`NoisyNetworks` into a single network that also calculates noise covariance matrices. Each of the individual
+multiport :class:`NoisyNetworks` are added to a :class:`MultiNoisyNetworkSystem` object via the :func:`MultiNoisyNetworkSystem.add`
+method. These networks are then connected together via the :func:`MultiNoisyNetworkSystem.connect` method. A new
+:class:`NoisyNetwork` is then created out of the collection of networks by calling :func:`MultiNoisyNetworkSystem.reduce`
+
+See scikit-rf examples on MultiNoisyNetworkSystem
+
+
+MultiNoisyNetworkSystem Class
+=============================
+
+.. autosummary::
+    :toctree: generated/
+
+    MultiNoisyNetworkSystem
+
+Building a Network from Multiple Networks
+=========================================
+
+.. autosummary::
+    :toctree: generated/
+
+    MultiNoisyNetworkSystem.add
+    MultiNoisyNetworkSystem.connect
+    MultiNoisyNetworkSystem.external_port
+    MultiNoisyNetworkSystem.verify
+    MultiNoisyNetworkSystem.reduce
+
+Extras
+============================
+
+.. autosummary::
+    :toctree: generated/
+
+    MultiNoisyNetworkSystem.num_networks
+
+"""
+
+import numpy as npy
+from scipy.linalg import block_diag
+from .noisyNetwork import NoisyNetwork
+from .networkNoiseCov import NetworkNoiseCov
+
+class MultiNoisyNetworkSystem(object):
```

Comment:
Before adding a new class to the skrf API (which will be a future source of issues ;), I'm asking myself "it is necessary?" 

If I understand well, a `MultiNoisyNetworkSystem ` , If only made of 2ports, consists in a multiple cascaded NoisyNetworks, am I right ? 
What if some NoisyNetwork in the list have more ports than 2 ? 
It sounds like a kind of `Circuit` for NoisyNetworks

Reply:
Yeah, see my above comment. MultiNoisyNetworkSystem works with multi-port NoisyNetworks. So more ports than two-ports. I made a lousy example that only worked with two-ports, I should have made something with splitters and circulators or somethings. 

This is very much like Circuit for NoisyNetworks.

Reply:
OK, I understand. 

For the sake of clarity, why not calling it `NoisyCircuit` then ? OK, the way of defining such circuit is a bit different from Circuit's way, but the purpose looks like identical.

From the theoretical point of, it would be interesting to see if the Circuit's building algorithm taken from (Hallbjörner, P., 2003, . Microw. Opt. Technol. Lett. 38, 99–102. https://doi.org/10/d27t7m ) could be extended to noisy Network as defined in Garcia's 2004 paper.